documentation/src/docs/controller-service.md

```diff
@@ -154,296 +151,276 @@ Controller Process Diagram
+ 3. Successor queries are performed on a single Stream Segment whereas truncation workflows work on a group of Stream Segments.
+
+ This ensures that during truncation we are able to retrieve sealed sizes for multiple Stream Segments with minimal number of calls into underlying metadata store. Since we could have arbitrarily large number of Stream Segments that have been sealed away, we cannot store all of the information in a single map and hence we shard the map and store it. The sharding function we use is to hash the creation epoch and get the shard number.
 
 The following are the Transaction Related metadata records:
 
-   - **Active Transactions**: Each new Transaction is created under this znode. This stores metadata
- corresponding to each Transaction as *ActiveTransactionRecord*. Once a
+   - **Active Transactions**: Each new Transaction is created under the znode. This stores metadata
+ corresponding to each Transaction as *Active Transaction Record*. Once a
  Transaction is completed, a new node is created under the global
- Completed Transaction znode and removed from under the stream specific
- _active_ Transaction node.
+ _Completed Transaction_ znode and removed from under the Stream specific
+ _Active Transaction_ node.
 
-   - **Completed Transactions**: All completed transactions for all streams are moved under this single
+   - **Completed Transactions**: All completed transactions for all Streams are moved under this single
  znode upon completion (via either commit or abort paths). We can
  subsequently garbage collect these values periodically following any
  collection scheme we deem fit. We have not implemented any scheme at
  this point though.
 
 ### Stream Store Caching
 
-Since there could be multiple concurrent requests for a given stream
+#### In-memory Cache
+Since there could be multiple concurrent requests for a given Stream
 being processed by same Controller instance, it is suboptimal to read
 the value by querying Zookeeper every time. So we have introduced an
-in-memory cache that each stream store maintains. It caches retrieved
-metadata per stream so that there is maximum one copy of the data per
-stream in the cache. There are two in-memory caches: _A cache of
-multiple stream objects in the store_ and _cache properties of a stream in
-the stream object_.
-
-We have introduced a concept of operation context and at the start of
-any new operation a new operation context is created. The creation of a
-new operation context invalidates the cached entities for a stream and
+**in-memory cache** that each Stream store maintains. It caches retrieved
+metadata per Stream so that there is maximum one copy of the data per
+Stream in the cache. There are two in-memory caches:
+
+- _A cache of
+multiple Stream objects in the store_
+- _Cache properties of a Stream in
+the Stream object_.
+
+#### Operation Context
```

Comment:
If this sub-section is the explanation of why it is not an issue to cache, then let's make the connection clearly in the text.

Reply:
Sub-section changed to bulleted point